### PR TITLE
Solved global exception handler and data validation

### PR DESCRIPTION
**Pull request by Krasnov-Maksim**

### CODE REVIEW

File: src/main/java/mate/academy/bookstore/validation/CreateBookRequestDtoValidator.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import mate.academy.bookstore.validation.annotation.CreateBookRequestDtoValidation;
+
+public class CreateBookRequestDtoValidator
+        implements ConstraintValidator<CreateBookRequestDtoValidation, CreateBookRequestDto> {
+    @Override
+    public boolean isValid(CreateBookRequestDto value, ConstraintValidatorContext context) {
+        return value.price() != null && value.price().doubleValue() > 0
+                && value.title() != null && !value.title().isBlank()
+                && value.author() != null && !value.author().isBlank();
+    }
```
Comment (sokimaaa) [inline src/main/java/mate/academy/bookstore/validation/CreateBookRequestDtoValidator.java]: you dont need this. 

you can use @NotNull, @PositiveOrZero and so on.

File: src/main/java/mate/academy/bookstore/dto/CreateBookRequestDto.java [hunk 2]
```diff
@@ -1,11 +1,18 @@
...
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.PositiveOrZero;
 import java.math.BigDecimal;
 
 public record CreateBookRequestDto(
+        @NotBlank()
         String title,
+        @NotBlank
         String author,
         String isbn,
```
Comment (Sanderfuck) [inline src/main/java/mate/academy/bookstore/dto/CreateBookRequestDto.java:15]: Valid isbn

File: src/main/java/mate/academy/bookstore/dto/CreateBookRequestDto.java [hunk 3]
```diff
@@ -1,11 +1,18 @@
...
+        @NotNull
+        @PositiveOrZero
         BigDecimal price,
         String description,
```
Comment (Sanderfuck) [inline src/main/java/mate/academy/bookstore/dto/CreateBookRequestDto.java:19]: You can valid description

File: src/main/java/mate/academy/bookstore/exception/CustomGlobalExceptionHandler.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+            MethodArgumentNotValidException ex,
+            HttpHeaders headers,
+            HttpStatusCode status,
+            WebRequest request
+    ) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        List<Object> errors = ex.getBindingResult().getAllErrors().stream()
+                .map(this::getErrorMessage)
+                .toList();
+        body.put("errors", errors);
```
Comment (Sanderfuck) [inline src/main/java/mate/academy/bookstore/exception/CustomGlobalExceptionHandler.java]: Why you don`t add timestamp?

**Review by Sanderfuck — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/bookstore/dto/CreateBookRequestDto.java (hunk 3 above)
Comment (Sanderfuck) [reply src/main/java/mate/academy/bookstore/dto/CreateBookRequestDto.java:19]: You can valid more, by size, etc

File: src/main/java/mate/academy/bookstore/dto/CreateBookRequestDto.java (hunk 2 above)
Comment (Sanderfuck) [reply src/main/java/mate/academy/bookstore/dto/CreateBookRequestDto.java:15]: You can create your own validation annotation for isbn

File: src/main/java/mate/academy/bookstore/dto/CreateBookRequestDto.java [hunk 5]
```diff
@@ -1,12 +1,21 @@
...
 import java.math.BigDecimal;
 
 public record CreateBookRequestDto(
+        @NotBlank()
         String title,
+        @NotBlank
         String author,
+        @NotBlank
         String isbn,
+        @NotNull
```
Comment (Sanderfuck) [inline src/main/java/mate/academy/bookstore/dto/CreateBookRequestDto.java]: Do you need not null here anymore?

**Review by Sanderfuck — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/bookstore/validation/IsbnValidator.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
+package mate.academy.bookstore.validation;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import java.util.regex.Pattern;
+
+public class IsbnValidator implements ConstraintValidator<Isbn, String> {
+    private static final Pattern NOT_DIGITS = Pattern.compile("[\\D]");
+    private static final int Isbn10Length = 10;
+    private static final int Isbn13Length = 13;
```
Comment (koval-kn) [inline src/main/java/mate/academy/bookstore/validation/IsbnValidator.java:10]: constants should be uppercased

**Review by koval-kn — APPROVED** [review]

**Pull request merged**